server/apiserver/argoserver.go

```diff
@@ -109,8 +113,16 @@ func init() {
 	}
 }
 
+func getResourceCacheNamespace(opts ArgoServerOpts) string {
```

Comment:
I think you always want to use opts.managedNamespace?

Reply:
I thought about it.
Case 1
- Installed argo in cluster mode
- In this case, we need to look at service accounts of multiple namespaces

Case 2
- Installed argo in namespace mode
- In this case, we need to see service accounts of installed namespace (typically `argo`)

Case 3
- Installed argo in managed namespace mode
- So, in this case say, we use `argo` namespace for installation, and `user-namespace` for usage
- So, for login, I would still need to use service accounts of `argo` namespace

Since for case 2 and 3, we only need service accounts of a single namespace,
we will just use the installation namespace

Reply:
Case 1:

cacheNamespace  == managedNamespace == "" == NamespaceAll
namespace == "argo"

Case 2:

cacheNamespace  == managedNamespace == namespace == "argo"

Case 3:

managedNamespace == userNamespace
namespace == "argo"
cacheNamespace == ???

So it is case 3 that is odd. It should be whatever it is today, which I think is "argo", which seems wrong to me. I would actually think you'd want the user namespace. But history is history.

Lets make this overridable with config, so if we are wrong, then we can change out mind.

Maybe call it `ssoNamespace`?

Reply:
ok, we can use `ssoNamespace`
so for case 1 we will set `ssoNamespace` = NamespaceAll
For Case 2 and 3 we will set `ssoNamespace` = namespace (typically argo)
Will make these changes